nautobot/utilities/forms/fields.py

```diff
@@ -666,3 +666,41 @@ def to_python(self, value):
         except ValueError as error:
             raise ValidationError(error)
         return super().to_python(value)
+
+
+class NaturalKeyMultipleChoiceField(django_filters.fields.ModelMultipleChoiceField):
```

Comment:
I feel like `NaturalKeyMultipleChoice...` overpromises as a name. Thoughts on something like `SlugOrPKMultipleChoice...` instead?

Reply:
NaturalKey might be confusing as well. I went with `SlugOrPKMultipleChoiceFilter` and `SlugOrPKMultipleChoiceField`